src/NodeVisitor/EscaperNodeVisitor.php

```diff
@@ -196,7 +196,7 @@ private function isSafeFor(string $type, AbstractExpression $expression, Environ
             $safe = $this->safeAnalysis->getSafe($expression);
         }
 
-        return \in_array($type, $safe) || \in_array('all', $safe);
+        return null !== $safe && (\in_array($type, $safe) || \in_array('all', $safe));
```

Comment:
Even if it's never `null` with core extensions, it's a possibility that we need to take into account.